pkg/kube/fake_pod_controller.go

```diff
@@ -0,0 +1,111 @@
+package kube
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type FakeKubePodController struct {
+	Podname string
+	PodObj  *corev1.Pod
+
+	StartPodCalled bool
+	StartPodErr    error
+
+	WaitForPodReadyCalled bool
+	WaitForPodReadyErr    error
+
+	// getCommandExecutorCalled bool
+	GetCommandExecutorRet PodCommandExecutor
+	GetCommandExecutorErr error
+
+	GetFileWriterCalled bool
+	GetFileWriterRet    *FakeKubePodFileWriter
+	GetFileWriterErr    error
+
+	StopPodCalled        bool
+	StopPodErr           error
+	InStopPodStopTimeout time.Duration
+	InStopPodGracePeriod int64
+}
+
+func (fkpr *FakeKubePodController) Pod() *corev1.Pod {
```

Comment:
nit: `fkpr` probably should be renamed either to `fkpc` or `pc`

Reply:
done